src/components/FilterBarInline/FilterBarInline.tsx

```diff
@@ -0,0 +1,159 @@
+import { chakra, ChakraComponent, Stack } from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+
+import Button from "../Button/Button";
+import ButtonGroup from "../ButtonGroup/ButtonGroup";
+import { LayoutTypes } from "../../helpers/types";
+import {
+  MultiSelectWidths,
+  SelectedItems as MultiSelectItems,
+} from "../MultiSelect/MultiSelect";
+import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";
+import ComponentWrapper from "../ComponentWrapper/ComponentWrapper";
+import Heading, { HeadingSizes } from "../Heading/Heading";
+
+export type FilterBarItemsType =
+  | (boolean | number | number[] | string | string[] | MultiSelectItems)[];
+
+interface FilterBarInlineProps {
+  /** The className of the FilterBarInline. */
+  className?: string;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Optional string value used to set the text for a `Heading` component, or
+   * a DS Heading component that can be passed in.
+   */
+  heading?: string | JSX.Element;
+  /** Renders the layout of the FilterBarInline child components in a row or column. */
+  layout?: LayoutTypes;
+  /** Function for the global `Clear Filters` button. If passed the `Clear Filters` button will render. */
+  onClear?: () => void;
+  /** Function for the global `Apply Filters` button. If passed the `Apply Filters` button will render. */
+  onSubmit?: () => void;
+  /** Render prop function used to render child components with updated props. */
+  renderChildren: ({
+    isBlockElement,
+    layout,
+    width,
+  }: {
+    isBlockElement?: boolean;
+    layout?: LayoutTypes;
+    width?: MultiSelectWidths;
+  }) => React.ReactNode;
+  /** The selected items state (items that were checked by user). */
+  selectedItems?: FilterBarItemsType;
+}
+
+/**
+ * `FilterBarInline` is a wrapper component for filter components and individual form input components.
+ * The DS considers the following components as filter components: `CheckboxGroup`, `MultiSelectGroup`,
+ * `RadioGroup`, `Checkbox`, `DatePicker`, `MultiSelect`, `Radio`, `Select`, `Slider`, `TextInput`, and `Toggle`.
+ *
+ * The wrapped components/component groups can be displayed in a column or
+ * in a row layout. `FilterBarInline` can render additional `Clear All` and a `Apply Filters` buttons. The two
+ * _optional_ buttons are controlled by the `onClear` or `onSubmit` props respectively.
```

Comment:
I think it could be clearer here that using these buttons is not the officially recommended pattern (if that's the final call)